### PR TITLE
FQT: RFQT, more rollups, etc

### PR DESCRIPTION
**Pull request by dorothy-zbornak**
- Add RFQT taker enforcement to FQT.
  - Any order where `takerAddress == FlashWallet` is considered an RFQT order.
  - We only fill RFQT orders if:
    - `context.callDataHash != bytes32(0)` (calldata is signed)
    - `data.rfqtTakerAddress != address(0) || context.taker == data.rfqtTakerAddress`
      - This buys us time to work on properly propagating the `rfqtTakerAddress` from 0x-API down to AS. We will just set `data.rfqtTakerAddress = 0` for the moment.
- Remove redundant event from `BridgeAdapter`.
- Treat all calldata as signed if quote signer is not set in `TransformERC20Feature`.
  - This is to buy us time getting 0x-API signing infra up without breaking RFQT as-is.
- Rollup the new Kyber bridge logic.
- Rollup Mooniswap.
- Integration tests for EP + MTX + RFQT
- Cosmetic refactors of the bridge mixins for consistency.
- We now return the mooniswap pool in the sampler, encode it in the bridge data, and decode it in the Mooniswap bridge adapter.

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/zero-ex/contracts/src/features/TransformERC20Feature.sol [hunk 1]
```diff
@@ -386,13 +386,19 @@ contract TransformERC20Feature is
         view
         returns (bytes32 validCallDataHash)
     {
+        address quoteSigner = getQuoteSigner();
+        if (quoteSigner == address(0)) {
+            // If no quote signer is configured, then all calldata hashes are
+            // valid.
+            return callDataHash;
```
Comment (dorothy-zbornak) [inline contracts/zero-ex/contracts/src/features/TransformERC20Feature.sol:396]: This is to buy us time getting 0x-API signing infra up without breaking RFQT as-is

File: contracts/zero-ex/contracts/src/transformers/FillQuoteTransformer.sol [hunk 2]
```diff
@@ -178,9 +182,14 @@ contract FillQuoteTransformer is
         // Approve the ERC20 proxy to spend `sellToken`.
         data.sellToken.approveIfBelow(erc20Proxy, data.fillAmount);
 
-        // Fill the orders.
         state.protocolFee = exchange.protocolFeeMultiplier().safeMul(tx.gasprice);
         state.ethRemaining = address(this).balance;
+        // RFQT orders can only be filled if we have a valid calldata hash
+        // (calldata was signed), and the actual taker matches the RFQT taker (if set).
+        state.isRfqtAllowed = context.callDataHash != bytes32(0)
+            && (data.rfqtTakerAddress == address(0) || context.taker == data.rfqtTakerAddress);
```
Comment (dorothy-zbornak) [inline contracts/zero-ex/contracts/src/transformers/FillQuoteTransformer.sol:190]: This buys us time to work on properly propagating the `rfqtTakerAddress` from 0x-API down to AS. We will just set `data.rfqtTakerAddress = 0` for the moment.

File: packages/asset-swapper/src/quote_consumers/exchange_proxy_swap_quote_consumer.ts [hunk 3]
```diff
@@ -118,6 +118,7 @@ export class ExchangeProxySwapQuoteConsumer implements SwapQuoteConsumerBase {
                     refundReceiver: refundReceiver || NULL_ADDRESS,
                     fillAmount: firstHopOrder.takerAssetAmount,
                     maxOrderFillAmounts: [],
+                    rfqtTakerAddress: NULL_ADDRESS,
```
Comment (dorothy-zbornak) [inline packages/asset-swapper/src/quote_consumers/exchange_proxy_swap_quote_consumer.ts:121]: NULL for now.

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinMooniswap.sol [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    function _tradeMooniswap(
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount,
+        bytes memory bridgeData
+    )
+        internal
+        returns (uint256 boughtAmount)
+    {
+        (IERC20TokenV06 sellToken, IMooniswapPool pool) =
+            abi.decode(bridgeData, (IERC20TokenV06, IMooniswapPool));
```
Comment (dorothy-zbornak) [inline contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinMooniswap.sol:69]: We now encode the pool as a second parameter in the bridge data. `MooniswapBridge` just does `abi.decode(bridgeData, (address))` but it should still work for both.

File: contracts/zero-ex/test/features/transform_erc20_test.ts [hunk 5]
```diff
@@ -28,11 +28,11 @@ import {
     TransformERC20FeatureEvents,
 } from '../wrappers';
 
-const { NULL_BYTES, NULL_BYTES32 } = constants;
+const { NULL_ADDRESS, NULL_BYTES, NULL_BYTES32 } = constants;
 
 type MintTokenTransformerEvent = DecodedLogEntry<TestMintTokenERC20TransformerMintTransformEventArgs>;
 
-blockchainTests.resets('TransformERC20 feature', env => {
+blockchainTests.resets.only('TransformERC20 feature', env => {
```
Comment (moodlezoup) [inline contracts/zero-ex/test/features/transform_erc20_test.ts]:
```suggestion
blockchainTests.resets('TransformERC20 feature', env => {
```

**Review by moodlezoup — APPROVED** [review]
one stray `.only`, looks good otherwise

File: packages/contract-addresses/addresses.json [hunk 6]
```diff
@@ -38,15 +38,15 @@
...
         "mooniswapBridge": "0x02b7eca484ad960fca3f7709e0b2ac81eec3069c",
         "transformers": {
-            "wethTransformer": "0x7bab5f7299e1ca123bb44eb71e6c89be7e558cc8",
-            "payTakerTransformer": "0xe8c07a119452b55eee2f999478aab97f3656d841",
-            "fillQuoteTransformer": "0xa8c8cf29699f223766f47fe79e2b7eb1a90e08c8",
-            "affiliateFeeTransformer": "0x9d7174f55b50dad2e417bd567ad2da1ae4eef76d"
+            "wethTransformer": "0x9fbacfe5cf2ee08a3f7c71c638603b6d5961bfeb",
+            "payTakerTransformer": "0xc9406e3ec8c4005951e0d9430a2981ca720d76c9",
+            "affiliateFeeTransformer": "0x2cfdacce1dcf97950a7dd0adad278a1077fa4224",
+            "fillQuoteTransformer": "0x3ae59e53ea6a2e364ba68e584c8db95134a0021a"
```
Comment (dorothy-zbornak) [inline packages/contract-addresses/addresses.json]: Sorted these according to order of deployment (increasing nonce).

**Review by hysz — APPROVED** [review]
lgtm!

**Review by dekz — APPROVED** [review]

**Pull request merged**